### ISSUE TITLE
Train and expose baseline churn model via CLI, saving metrics/artifacts

### ISSUE THREAD

**Issue by jbrdge** [jbrdge/churn-prediction#28]
## Description
Build a minimal, reproducible baseline model to predict churn. Provide a CLI entry point to train/evaluate, and persist artifacts (model, metrics, plots) for inspection and future comparison. This depends on v0.4.0 ETL outputs.

## Acceptance Criteria
- [ ] **CLI command** exists (e.g., `python -m src.cli.train_baseline ...`) with flags for input path, target column, test size, random seed, and output directory.
- [ ] **Model:** simple baseline (e.g., Logistic Regression) trained on v0.4.0 features/labels.
- [ ] **Artifacts saved** under a versioned folder (e.g., `artifacts/baseline_v1/`):
    - `model.pkl` (serialized model)
    - `metrics.json` (accuracy, precision, recall, F1, ROC-AUC)
    - `params.json` (hyperparameters + seed)
    - `coefficients.csv` or `feature_importances.csv`
    - `confusion_matrix.png` and `roc_curve.png`
- [ ] **Reproducibility:** fixed random seed; add `scikit-learn` and `matplotlib` to `requirements.txt`.
- [ ] **Makefile target:** `make train-baseline` that runs the CLI with sensible defaults.
- [ ] **README:** “Baseline Model” section with one-line quickstart and artifact descriptions.
- [ ] **Light tests:** `tests/test_baseline_cli.py` creates a tiny synthetic dataset and verifies CLI produces expected files (no accuracy expectations yet).
- [ ] **.gitignore updates:** ignore large artifacts by default but keep a tiny example (e.g., `artifacts/.gitkeep`).

## Example CLI (document in README):
```bash
python -m src.cli.train_baseline \
  --input data/processed/features.parquet \
  --target churned \
  --test-size 0.2 \
  --random-state 42 \
  --outdir artifacts/baseline_v1
```

## Definition of Done:
- Single command trains and produces documented artifacts without manual steps.
- Artifacts and metrics are saved deterministically (given the seed).
- README instructions allow another developer to reproduce the run.
